app/v1/models/errors/mtdErrors.scala

```diff
@@ -36,6 +36,14 @@ object RuleIncorrectOrEmptyBodyError extends MtdError("RULE_INCORRECT_OR_EMPTY_B
 object RuleTaxYearRangeExceededError
     extends MtdError("RULE_TAX_YEAR_RANGE_EXCEEDED", "Tax year range exceeded. A tax year range of one year is required.")
 
+object RuleTypeOfLossUnsupported extends MtdError("FORMAT_TYPE_OF_LOSS", "The supplied type of loss format is invalid")
+
+object RuleInvalidSelfEmploymentId extends MtdError("FORMAT_SELF_EMPLOYMENT_ID", "The supplied self employment ID format is invalid")
+
+object RulePropertySelfEmploymentId extends MtdError("RULE_SELF_EMPLOYMENT_ID", "An ID was supplied for a non-self employment business type")
+
+object RuleInvalidLossAmount extends MtdError("RULE_LOSS_AMOUNT", "Amount should be a positive number less than 99999999999.99 with up to 2 decimal places")
```

Comment:
Missing `FORMAT_LOSS_AMOUNT`

Reply:
I am confused as to how this error code can ever be handled at a validator level. The formatting has already been applied at the json level and I can do no checks against a BigDecimal which would ever produce an error for formatting.

Reply:
I don't get it either tbh. Maybe it would work if the `amount.scale < 3` part of `AmountValidation` was taken out and moved into its own validator? I guess checking the number of decimal places doesn't really make sense in an "is this value between 0 and 99999999999.99" validator. Might be worth asking Mark when he gets in though.

Reply:
This has now been updated as per discussions